### PR TITLE
Implement a cached Block class that minimizes DOM access

### PR DESCRIPTION
**Pull request by frostburn**

### CODE REVIEW

File: lib/ui/cachedblock.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const Block = require('./block');
```
Comment (RauliL) [inline lib/ui/cachedblock.js]: With Webpack and Babel now installed, you can use
```JavaScript
import Block from './block';
import { blockSize } from './variables';
```
instead which is the ES6 way of doing things.

File: lib/ui/cachedblock.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+const { blockSize } = require('./variables');
+
+class CachedBlock extends Block {
```
Comment (RauliL) [inline lib/ui/cachedblock.js]: Why another class? Can't these be implemented into `Block` itself?

Comment (frostburn) [inline lib/ui/cachedblock.js]: Sure, but with React or re:dom coming up we might just want to deprecate this class soon.

File: lib/ui/cachedblock.js (hunk 1 above)
Comment (frostburn) [inline lib/ui/cachedblock.js]: Cool. Should be used globally though so it needs a separate PR.

File: lib/ui/cachedblock.js (hunk 2 above)
Comment (RauliL) [inline lib/ui/cachedblock.js]: I don't think `Block` class is going anywhere especially if we use RE:DOM. It just requires some tweaking.

Comment (frostburn) [inline lib/ui/cachedblock.js]: But this caching stuff doesn't belong there. It might actually break things with a separate DOM engine.

**Pull request merged**